doc/manual.rst

```diff
@@ -5241,15 +5241,21 @@ chance to convert it into a sequence.
 Macros
 ======
 
-A macro is a special kind of low level template. Macros can be used
-to implement `domain specific languages`:idx:.
+A macro is a special function that is executed on the virtual machine
+of the Compiler. Normally the input for a macro is an abstract syntax
+tree (AST) of the code that is passed to it. The macro can then do
+transformations on it and returned the now transformed AST. The
+transformed AST is then passed to the compiler as if the macro
```

Comment:
> The transformed AST is then passed to the compiler as if the macro invocation would have been replaced by its result in the source code.

This sentence needs to be reworded. No idea to what though :)

Reply:
Me neither. I think it is rather short.